### PR TITLE
Shopper can place an order as logged in user

### LINKED ISSUE #5786: Critical flows: Shopper → Checkout → Can place an order as logged in user
### Story

As a shopper, I want to be able to place an order as a logged-in user so that I can look up my order later.

### File

`tests/e2e/specs/shopper/checkout-account.test.js`

### Precondition

Ensure that the shopper is logged in before running the test.

### Test

1. Go to the product page.
2. Add the product to the cart.
3. Go to the checkout page.
4. Place the order.
5. Verify that the order had been placed successfully.

### PR DESCRIPTION
**Pull request by nielslange**
Fixes #5786

### Notes

This PR adds e2e tests to verify that a shopper can place an order while being logged in. 

#### Automated Tests
* [x] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [x] E2E tests

### Manual Testing

1. Check out this PR.
2. Start Docker Desktop.
3. Run `npm run wp-env destroy && npm run wp-env start` to start a fresh e2e environment.
4. Run `npm run test:e2e -- tests/e2e/specs/shopper/checkout-place-order.test.js` to run only this test case.
5. Run `npm run test:e2e` to run the whole test suite.

### CODE REVIEW

File: tests/e2e/specs/shopper/checkout-place-order.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		await shopper.block.placeOrder();
+		await expect( page ).toMatch( 'Your order has been received.' );
+	} );
+
+	it( 'allows customer to place an order as a logged in user', async () => {
+		await shopper.login();
+		await shopper.goToShop();
+		await shopper.addToCartFromShopPage( SIMPLE_PRODUCT_NAME );
+		await shopper.block.goToCheckout();
+		await shopper.block.placeOrder();
```
Comment (tarhi-saad) [inline tests/e2e/specs/shopper/checkout-place-order.test.js:28]: The second test does fail on my end. We need to fill in the billing details here as well: `await shopper.block.fillBillingDetails( BILLING_DETAILS );`

**Review by tarhi-saad — CHANGES_REQUESTED** [review]
Great job! 👍 Thanks, @nielslange! There is just a small issue to fix before we merge the changes. Please check the comment below.

File: tests/e2e/specs/shopper/checkout-place-order.test.js (hunk 1 above)
Comment (nielslange) [reply tests/e2e/specs/shopper/checkout-place-order.test.js:28]: Done! 😀

**Review by tarhi-saad — APPROVED** [review]
Thank you, @nielslange! Let's 🚢 it!

**Pull request merged**